ClassLibrary/ClassLibrary/Library.cs

```diff
@@ -41,23 +45,51 @@ public List<Author> Authors
         } 
             = new List<Author>();
 
-        public Author AddAuthor(String nameAuthor,String lastNameAuthor)
+        public void AddAuthorInList(Author author)
         {
-            if (nameAuthor.IsNullOrWhiteSpace() || lastNameAuthor.IsNullOrWhiteSpace())
+            if (author == null)
             {
-                throw new NullReferenceException("The author's first or last name is empty ");
+                throw new NullReferenceException("Author's reference is null");
             }
 
-            Author returnAuthor = Authors?.FirstOrDefault(author =>
-                author.Name == nameAuthor && author.LastName == lastNameAuthor);
+            Author returnAuthor = Authors?.FirstOrDefault(x =>
+                x.ID == author.ID);
 
             if (returnAuthor == null)
             {
-                returnAuthor = new Author(nameAuthor, lastNameAuthor);
-                Authors.Add(returnAuthor);
+                Authors.Add(author);
+            }
+            else
+            {
+                throw new ArgumentException("This author exists in List");
```

Comment:
better to say `$"Author [{author.ID}] {author.Name} {author.LastName} already added to the library"`